[Doc] kerberos debug instructions (#15779)
elastic/logstash: docs/static/troubleshoot/ts-kafka.asciidoc

* If (and only if) upgrading is not possible, set `consumer_threads` to `1`.

[discrete] 
[[ts-kafka-kerberos-debug]]
===== Debug Kerberos SASL

> To troubleshoot authentication failure in Kafka client, add `-Dsun.security.krb5.debug=true` to `jvm.options` and the following lines to `log4j2.properties`

Reviewer: ```suggestion
To troubleshoot authentication failures in Kafka client, add `-Dsun.security.krb5.debug=true` to `jvm.options` and the following lines to `log4j2.properties`:
```

```

> logger.kafkainput.name = logstash.inputs.kafka
> logger.kafkainput.level = debug

Reviewer: ```suggestion
logger.kafkainput.name = logstash.inputs.kafka
logger.kafkainput.level = debug
logger.kafkaoutput.name = logstash.outputs.kafka
logger.kafkaoutput.level = debug
```

logger.kafka.name = org.apache.kafka
logger.kafka.level = debug
```

> NOTE: Kerberos debug log is not being output through log4j but is directed to the console.

Reviewer: ```suggestion
NOTE: Logging entries for Kerberos are NOT sent through Log4j but go directly to the console.
```

* If (and only if) upgrading is not possible, set `consumer_threads` to `1`.

> [discrete] 
> [[ts-kafka-kerberos-debug]]
> ===== Debug Kerberos SASL
> 
> To troubleshoot authentication failures in Kafka client, add `-Dsun.security.krb5.debug=true` to `jvm.options` and the following lines to `log4j2.properties`:
> 
> ```
> logger.kafkainput.name = logstash.inputs.kafka
> logger.kafkainput.level = debug
> logger.kafkaoutput.name = logstash.outputs.kafka
> logger.kafkaoutput.level = debug
> logger.kafka.name = org.apache.kafka
> logger.kafka.level = debug
> ```
> 
> NOTE: Logging entries for Kerberos are NOT sent through Log4j but go directly to the console.

Reviewer 2: ```suggestion
[discrete] 
[[ts-kafka-kerberos-debug]]
===== Debug Kerberos SASL

You can set up your machine to help you troubleshoot authentication failures in the Kafka client.

* In `config/jvm.options`, add:
+
[source,txt]
-----
`-Dsun.security.krb5.debug=true` to `jvm.options` 
-----

* In `config/log4j2.properties`, add: 
+
[source,txt]
-----
logger.kafkainput.name = logstash.inputs.kafka
logger.kafkainput.level = debug
logger.kafkaoutput.name = logstash.outputs.kafka
logger.kafkaoutput.level = debug
logger.kafka.name = org.apache.kafka
logger.kafka.level = debug
-----

NOTE: Logging entries for Kerberos are NOT sent through Log4j but go directly to the console.
```

Reviewer 2: Proposing a formatting tweak to make sure that the inline `jvm.options` setting doesn't get overlooked, and to make both settings more eye-catching.

Reviewer 2: Can we give users more info on next steps or what to look for? Right now, we're helping them set up debugging and then dropping them.

Author: We mainly want to write down how to enable debug log to check kerberos issues as it is tricky that the log does not send to log4j log. The next step is that user needs to get the hints from debug log. Do you think we need to add a line to remind them to check the log?

* If (and only if) upgrading is not possible, set `consumer_threads` to `1`.

[discrete] 
[[ts-kafka-kerberos-debug]]

> ===== Debug Kerberos SASL

Author: ```suggestion
===== Setting up debugging for Kerberos SASL
```

+
[source,txt]
-----

> -Dsun.security.krb5.debug=true

Reviewer 2: Thanks for catching and fixing this, @kaisecheng  :-D
